backend/dataall/modules/maintenance/services/maintenance_service.py

```diff
@@ -0,0 +1,119 @@
+"""
+A service layer for maintenance activity
+Defines functions and business logic to be performed for maintenance window
+"""
+
+import logging
+import os
+
+from dataall.base.aws.event_bridge import EventBridge
+from dataall.base.aws.parameter_store import ParameterStoreManager
+from dataall.modules.maintenance.api.enums import MaintenanceStatus
+from dataall.modules.maintenance.db.maintenance_repository import MaintenanceRepository
+from dataall.core.stacks.aws.ecs import Ecs
+
+logger = logging.getLogger(__name__)
+
+
+class MaintenanceService:
+    # Update the RDS table with the mode and status to PENDING
+    # Disable all scheduled ECS tasks which are created by data.all
+    @staticmethod
+    def start_maintenance_window(engine, mode: str = None):
+        logger.info('Putting data.all into maintenance')
+        try:
+            with engine.scoped_session() as session:
+                maintenance_record = MaintenanceRepository(session).get_maintenance_record()
+                if (
+                    maintenance_record.status == MaintenanceStatus.PENDING.value
+                    or maintenance_record.status == MaintenanceStatus.ACTIVE.value
+                ):
+                    logger.error(
+                        'Maintenance window already in PENDING or ACTIVE state. Cannot start maintenance window. Stop the maintenance window and start again'
+                    )
+                    return False
+                MaintenanceRepository(session).save_maintenance_status_and_mode(
+                    maintenance_status=MaintenanceStatus.PENDING.value, maintenance_mode=mode
+                )
+            # Disable scheduled ECS tasks
+            # Get all the SSM Params related to the scheduled tasks
+            ecs_scheduled_rules = ParameterStoreManager.get_parameters_by_path(
```

Comment:
nit: lines 40-46 are duplicated in 69-75, maybe we can create a _get_ecs_rules()

Reply:
Updated in the next commit. Thanks!